### DISCUSSION TITLE
Github Linker with Last Updated, Stars, etc...

### DISCUSSION THREAD

**Discussion by patrickcurl** [SilentVoid13/Templater#544] [Templates Showcase]
This is a nifty ...albeit a little ugly/hackety script to basically paste github links w/ extra meta data. It scrapes the github page for stars, description, user/repo, and last update time. I use this for package lists, personal 'awesome-like' lists so I know I'm using packages that are still being maintained, updated, etc.... Helps when I prune my lists too.

Format looks like:

**[GitHub - kmaasrud / awesome-obsidian](https://github.com/kmaasrud/awesome-obsidian)** - *🕶️ Awesome stuff for Obsidian* <br>(2.4k ⭐'s) | Sat, Jan 15, 2022 1:05 PM

there's the template file and the script... 

git_linker.js

```
async function git_linker(tp){
  const cb = await tp.system.clipboard();
  if (cb.includes('github.com')){
    // cb.includes('github.com');
    const link = cb;
  const page = await fetch(link).then(data => data.text());
  const p = new DOMParser();
  const doc = p.parseFromString(page, "text/html");
  const $ = s => doc.querySelector(s);
  const parts = $("meta[property='og:title']").getAttribute('content').split(':');
  const description = parts.splice(1).join(':').trim();
  const repoParts = parts[0].trim().split('/');
  const user = repoParts[0] || '';
  const repo = repoParts[1] || '';
  const starElement = $("span[id='repo-stars-counter-star']");
  const starCount =  starElement.innerText.trim() || starElement.textContent || 0;
  const stars = `${starCount} ⭐'s`;
  const lastUpdate = $('relative-time').getAttribute('datetime');
  const formattedDate = window.moment(lastUpdate).format('llll');

    return `**[${user} / ${repo}](${link})** - *${description}* <br>(${stars}) | ${formattedDate}`;
  } 

  return cb;

}
```

and the template file | GitLinker.md: 

```
<% tp.user.git_linker(tp) %>
```

**Comment by dredhorse** [comment]
the code doesn't work for me anymore on the main repo page.
const parts = $("meta[property='og:title']").getAttribute('content').split(':'); needs to be
const parts = $("meta[ property='og:title']").getAttribute('content').split(':');

const lastUpdate = $('relative-time').getAttribute('datetime'); returns null

on the wiki page const lastUpdate = $('relative-time').getAttribute('datetime'); returns the last updated time though..
